### PR TITLE
RT4337: Crash in DES

### PR DESCRIPTION
**Pull request by richsalz**
Salt must be two ASCII characters.  Add tests to check for that,
and a test to test the checks.

### CODE REVIEW

**Comment by richsalz** [comment]
This replaces #738.  Ping @mattcaswell , @kroeckx

**Comment by mattcaswell** [comment]
Probably we need to update the DES_fcrypt/DES_crypt documentation to point out the change in behaviour between 1.1.0 and earlier versions.

**Comment by richsalz** [comment]
updated commit with new documentation.

**Comment by mattcaswell** [comment]
@richsalz - did you push the update?

**Comment by richsalz** [comment]
oops, gitlab not github :)  pushed.

File: doc/crypto/des.pod [hunk 1]
```diff
@@ -272,22 +273,25 @@ DES_string_to_key() is available for backward compatibility with the
...
 Applications should use the higher level functions
 L<EVP_EncryptInit(3)> etc. instead of calling these
 functions directly.
 
 Single-key DES is insecure due to its short key size.  ECB mode is
 not suitable for most applications; see L<des_modes(7)>.
 
+=head1 HISTORY
+
+The requirement that the salt be two ASCII characters was enforced in
```
Comment (mattcaswell) [inline doc/crypto/des.pod]: This paragraph should mention that it refers to the DES_crypt() and DES_fcrypt() functions.

**Comment by mattcaswell** [comment]
+1 subject to the documentation amendment being made.

**Comment by richsalz** [comment]
Those are the only two functions that have a salt, but I'll future-proof the doc and add a phrase.  Thanks.

**Comment by richsalz** [comment]
commit 6493e48, thanks!

**Pull request closed without merging**